src/familiar.ts

```diff
@@ -281,3 +384,182 @@ export function timeToMeatify(): boolean {
+
+  const nominalOutfitValue = dropFamiliars
+    .map((fam) => {
+      const { meat, bonus } =
+        cachedOutfits.get(fam.leprechaunMultiplier) ??
+        cacheOutfit(fam.leprechaunMultiplier, fam.familiar);
+      return (meat * locBaseMeat) / 100 + bonus;
+    })
+    .reduce((a, b) => {
+      return a < b ? a : b;
+    });
+
+  let nonJellyExpectedAdv = 0;
+  const meatFam = meatFamiliar();
+  const meatFamEV = sum(dropFamiliars, (fam) => {
+    return fam.familiar === meatFam ? fam.marginalValue - nominalOutfitValue : 0;
+  });
+  const shouldRunDropFams =
+    Math.round(
+      sum(
+        dropFamiliars.map((fam) => {
+          const rotFam = rotatingFamiliars.reduce((a, b) => (a.familiar === fam.familiar ? a : b));
+          const { meat, bonus } =
+            cachedOutfits.get(fam.leprechaunMultiplier) ??
+            cacheOutfit(fam.leprechaunMultiplier, fam.familiar);
+
+          const outfitValue = (meat * locBaseMeat) / 100 + bonus - nominalOutfitValue;
+          if (rotFam.familiar === fam.familiar) {
+            const pIdx = get(rotFam.pref);
+            const dropVal = garboValue(rotFam.drop);
+            let expectedAdventures = 0;
+            for (let idx = pIdx; idx < rotFam.expected.length; idx++) {
+              if (dropVal / rotFam.expected[idx] + outfitValue > meatFamEV) {
+                expectedAdventures += rotFam.expected[idx];
+              } else break;
+            }
+            nonJellyExpectedAdv += expectedAdventures;
+            return expectedAdventures;
+          } else if (fam.familiar === $familiar`Space Jellyfish`) {
+            const pIdx = get("_spaceJellyfishDrops");
+            const dropVal =
+              barf && myInebriety() <= inebrietyLimit() ? garboValue($item`stench jelly`) : 0;
+            let expectedAdventures = 0;
+            for (let idx = pIdx; idx < 5; idx++) {
+              if (dropVal / (idx + 1) + outfitValue > meatFamEV) {
+                expectedAdventures += idx + 1;
+              } else break;
+            }
+            if (dropVal / 20 + outfitValue > meatFamEV) expectedAdventures += Infinity;
+            return expectedAdventures;
+          } else {
+            return 0;
+          }
+        }),
+        (val) => {
+          return val;
+        }
+      )
+    ) >=
+    myAdventures() - globalOptions.saveTurns;
+
+  if (shouldRunDropFams) {
+    const jellyfishEV =
+      dropFamiliars.filter((fam) => fam.familiar === $familiar`Space Jellyfish`)[0].marginalValue -
+        nominalOutfitValue ?? 0;
+    const idx =
+      globalOptions.ascending &&
+      nonJellyExpectedAdv < 30 && // Assume 30 overdrunk barf adventures
+      myInebriety() <= inebrietyLimit() &&
+      jellyfishEV > meatFamEV &&
+      dropFamiliars[0].familiar !== meatFamiliar()
+        ? dropFamiliars.findIndex((fam) => {
+            fam.familiar === (jellyfishEV > meatFamEV ? $familiar`Space Jellyfish` : meatFam);
+          }) ?? 0
+        : 0;
```

Comment:
.findIndex returns -1 for invalid inputs, not undefifined

Also I have no idea what this is trying to do, this logic should be sliced out into a function instead of being done with a massive ternary

Reply:
dropFamiliars[0] gives us the best marginal familiar to run
However, if we are ascending, then we might not want to use the best marginal familiar (saving it instead for overdrunk turns). `idx` thus tries to skip the best marginal familiar if the following conditions are met
1) We must be ascending
2) The number of remaining marginal familiar turns (excluding jellyfish) we expect to run is < 30 (we will then save these ~30 marginal familiar turns to run when overdrunk)
3) We must not be overdrunk
4) ~~I forgot to remove the `jellyfishEV > meatFamEV` check here (it should only be checked in the next instance where you see it)~~

Now if the above conditions are satisfied, we want to skip all our remaining marginal familiars, and only consider either the jellyfish or robort. If jellyfishEV > meatFamEV, we get the index of the jellyfish, else we get the index of the robort. Our marginal familiar is then chosen with dropFamiliars[idx]

Reply:
This would be great as comments interspersed into a well-named function